docs/TerraToolkitUpgradeGuide-v3.0.0.md

```diff
@@ -0,0 +1,121 @@
+# Terra Toolkit Upgrade Guide v3.0.0
+This document will provide information on upgrading from Terra Toolkit 2.x to 3.0.0.
+
+If you use terra-dev-site please also refer to it's upgrade guide as it abstracts away from additional config that is required here.
+
+## Webpack 4
+Terra toolkit and it's dependencies have been upgraded to consume webpack 4. Terra toolkit is now intended to provide Webpack to the libraries that consume it.
+Users of Terra toolkit should not add a direct dependency to webpack in their package JSON after updating. We have made this change to provide a single point of upgrading webpack.
+
+Terra toolkit now provides a default webpack config. This config will need to be referenced by the app level webpack config to add in an entry (unless consuming terra-dev-site). We recommend using [webpack-merge](https://github.com/survivejs/webpack-merge) to combine configs.
+The default webpack config is now a function that will flex between production and development modes by passing in the -p flag when compiling with webpack. For more about webpack config as a function go [here](https://webpack.js.org/configuration/configuration-types/).
+If you are consuming this webpack config to create another webpack config, don't forget that you need to execute the function first.
+
+```javascript
+const defaultWebpackFunction = require('terra-toolkit/lib/webpack/webpack.config');
+
+const defaultWebpackConfig = defaultWebpackFunction(); // This will give you the default dev webpack config.
+
+defaultWebpackConfig.entry = {
+    'index': path.resolve(path.join(__dirname, '..', 'Index')),
+  };
+
+module.exports = defaultWebpackConfig;
+
+```
+
+Translation Aggregation has been added to the default webpack config. By default all of the terra supported locales are included. If you need to customize this list, see the I18N aggregation section below. To completely disable translation aggregation within the webpack build to add your own, you can pass the environment variable `--env.disableAggregateTranslations` to the webpack command.
+
+```bash
+webpack --config src/webpack/webpack.config --env.disableAggregateTranslations
+```
+
+The `webpack` command is available to applications consuming terra-toolkit.
+
+The major change with webpack4 plugins is that the mini-css-extract-plugin now replaces extract-text-webpack-plugin which is no longer maintained.
+
+## I18N aggregation
+In a previous release, the aggregation translations script (`tt:aggregate-translations`) was added to terra-toolkit. In this release a new configuration option has been added. You can specify the aggregate-translations configuration through a `terraI18n.config.js` file in your root project directory.
+
+### Example
+```javascript
+const fse = require('fs-extra');
```

Comment:
I would be helpful to add `// terrai18n.config` here and then add an example of calling the function that uses this config to clarify how to use it

Reply:
I added the comment, but you don't have to call any function with the config, it's auto discovered.